pylint/checkers/base.py

```diff
@@ -2588,10 +2588,78 @@ def _check_type_x_is_y(self, node, left, operator, right):
         self.add_message("unidiomatic-typecheck", node=node)
 
 
+class ModifiedIterationChecker(_BasicChecker):
```

Comment:
Please create a new file we're trying to limit the number of very big file in the codebase and this one is already gigantic :) Could be ``modified_iteration_checker.py``